### PR TITLE
Refactor compare version functions

### PR DESCRIPTION
**Pull request by varjolintu**
Refactors the compare version functions:
- Move the actual `compareVersion()` function to `global.js`.
- Rename `keepass.compareVersion()`  to `keepass.compareVersions()` , and allow it to take in an Array for multiple comparisons.
- Add unit tests.

### CODE REVIEW

File: keepassxc-browser/background/keepass.js [hunk 1]
```diff
@@ -945,25 +945,18 @@ keepass.updateDatabaseHashToContent = async function() {
...
-        return false;
-    }
-
-    // Handle beta/snapshot builds as stable version
-    const snapshot = '-snapshot';
-    const beta = '-beta';
-    if (current.endsWith(snapshot)) {
-        current = current.slice(0, -snapshot.length);
+// Expects an array of versions to compare
+keepass.compareVersions = function(versions, current, canBeEqual = true) {
```
Comment (droidmonkey) [inline keepassxc-browser/background/keepass.js:949]: compareVersion (the global function) and keepass.compareVersions (the class function) are way too similar to each other which makes it hard to read and debug which function you actually intend to use. Recommend different naming choices.

File: tests/global.spec.ts [hunk 2]
```diff
@@ -1,11 +1,27 @@
 import { test, expect } from '@playwright/test';
 import {
+    compareVersion,
     matchesWithNodeName,
     siteMatch,
     slashNeededForUrl,
     trimURL
 } from '../keepassxc-browser/common/global.js';
 
+test('Test compareVersion()', async ({ page }) => {
```
Comment (droidmonkey) [inline tests/global.spec.ts:10]: Need tests for keepass.compareVersions

**Review by droidmonkey — CHANGES_REQUESTED** [review]

File: tests/global.spec.ts (hunk 2 above)
Comment (varjolintu) [reply tests/global.spec.ts:10]: At this point functions in `keepass.js` cannot be tested due to different dependencies the file expects. One solution could be to move this function entirely to `global.js` and `keepass.js` just returns its value directly.

**Review by droidmonkey — APPROVED** [review]

File: tests/global.spec.ts (hunk 2 above)
Comment (droidmonkey) [reply tests/global.spec.ts:10]: Nah it's good as it is

**Pull request merged**